### PR TITLE
support external file refs

### PR DESCRIPTION
**Pull request by kobik**

### CODE REVIEW

**Comment by kobik** [comment]
@kibertoad, can you please review?

**Comment by kibertoad** [comment]
@kobik Sure, will do within a couple of hours.

File: src/utils/schemaUtils.js [hunk 1]
```diff
@@ -144,11 +147,39 @@ function getSchemaType(dereferencedSchema) {
...
+
+function getSchemaBasePath(pathOrSchema, options = {}) {
+    // always return basePath from options if exists
+    if (options.basePath) {
+        return options.basePath;
+    }
+
+    // in case a path to defintions file was given
+    if (typeof pathOrSchema === 'string') {
+        const fullPath = path.resolve(pathOrSchema).split('/');
```
Comment (kobik) [inline src/utils/schemaUtils.js]: Use OS separator instead of `/`

File: package.json [hunk 2]
```diff
@@ -42,13 +42,18 @@
...
+    "Kobi Carmeli",
+    "Gal Cohen"
+  ],
   "dependencies": {
     "ajv": "^6.10.2",
     "clone-deep": "^4.0.1",
     "decimal.js": "^10.2.0",
     "js-yaml": "^3.13.1",
-    "json-schema-deref-sync": "^0.10.1",
+    "json-schema-deref-sync": "github:cvent/json-schema-deref-sync",
```
Comment (kibertoad) [inline package.json]: Wouldn't it be safer to point to https://github.com/cvent/json-schema-deref-sync/releases/tag/v0.11.0 directly?

File: src/utils/schemaLoaders.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
+const fs = require('fs');
+const path = require('path');
+const jsyaml = require('js-yaml');
+
+const cwd = process.cwd();
+
+const file = function (refValue, options) {
```
Comment (kibertoad) [inline src/utils/schemaLoaders.js]: any reason to make `file` an arrow function and not just a function?
Also isn't using noun for a function name somewhat confusing?

File: test/openapi3/general/external-ref/file-refs.test.js [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+const chai = require('chai').use(require('chai-as-promised'));
+const path = require('path');
+const fs = require('fs');
+const jsyaml = require('js-yaml');
+const { expect } = chai;
+
+const schemaValidatorGenerator = require('../../../../src');
+
+describe('Loading defintions file with file refs', () => {
+    describe('calling buildSchemaSync with path to defintions file', () => {
```
Comment (kibertoad) [inline test/openapi3/general/external-ref/file-refs.test.js]: defintions -> definitions (couple of times in this file)

File: test/openapi3/general/external-ref/file-refs.test.js [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+        const swaggerPath = path.join(__dirname, 'file-refs.yaml');
+        const schema = schemaValidatorGenerator.buildSchemaSync(swaggerPath, {});
+        const validator = schema['/users'].post.body['application/json'];
+
+        it('should validate accroding to yaml file schema', () => {
```
Comment (kibertoad) [inline test/openapi3/general/external-ref/file-refs.test.js]: accroding -> according

File: package.json (hunk 2 above)
Comment (kobik) [reply package.json]: they still haven't released it, that's why this PR is labeled as draft

File: test/openapi3/general/external-ref/file-refs.test.js (hunk 5 above)
Comment (kobik) [reply test/openapi3/general/external-ref/file-refs.test.js]: fixed

File: test/openapi3/general/external-ref/file-refs.test.js (hunk 4 above)
Comment (kobik) [reply test/openapi3/general/external-ref/file-refs.test.js]: fixed

File: src/utils/schemaLoaders.js (hunk 3 above)
Comment (kobik) [reply src/utils/schemaLoaders.js]: changed

File: package.json (hunk 2 above)
Comment (kibertoad) [reply package.json]: @kobik I don't mean the npm release; AFAIR, you can target specific git tags on GH repo.

**Comment by kobik** [comment]
Yes, it's possible.
But I do prefer merging this change only after they publish it to npm.

**Comment by kibertoad** [comment]
@kobik I think it was published: https://www.npmjs.com/package/json-schema-deref-sync

**Pull request merged**